monai/data/dataset.py

```diff
@@ -754,6 +775,12 @@ def _load_cache_item(self, idx: int):
         return item
 
     def _transform(self, index: int):
+        if self.hash_as_key:
+            key = self.hash_func(self.data[index])
+            if key in self._cache_keys[: self.cache_num]:
+                # if existing in cache, get the index
+                index = self._cache_keys.index(key)
```

Comment:
this will have O(n) searching twice, instead of O(1) hash look up... perhaps we can have (after line 742)
```py
self._cache = dict(zip(self._cache_keys[: self.cache_num], self._fill_cache())
```
then here the dict cache will have fast access:
```py
key = self.hash_func(self.data[index])
if key in self._cache:
    index = key  # this should work with `data = self._cache[index]`
```

Reply:
Hi @wyli ,

Thanks for your sugestion.
So in this way, the `self._cache` will be `dict` in Hash mode and be `list` in the previous regular mode?
Or maybe let me change it to `dict` directly even for previous regular mode?

Thanks.

Reply:
yes, to me it makes sense to have the hash-based using dict and index-based using list

Reply:
OK, updated in the latest commit.

Thanks.